pkg/time/time.go

```diff
@@ -16,8 +16,37 @@ func Open(l *lua.State) {
 
 var timeLibrary = []lua.RegistryFunction{
 	{"now", now},
+	{"format", format},
 	{"sleep", sleep},
 	{"since", since},
+	{"add", add},
+}
+
+func format(l *lua.State) int {
+	epochNanoToFormat := int64(lua.CheckNumber(l, 1))
+	layout := lua.CheckString(l, 2)
+	zone := lua.CheckString(l, 3)
+
+	loc, err := time.LoadLocation(zone)
+	if err != nil {
+		lua.Errorf(l, err.Error())
+	}
+
+	unixTime := time.Unix(0, epochNanoToFormat)
+	timeInTimeZone := unixTime.In(loc)
+	l.PushString(timeInTimeZone.Format(layout))
+
+	return 1
+}
+
+func add(l *lua.State) int {
+	start := int64(lua.CheckNumber(l, 1))
+	startUnix := time.Unix(0, start)
+	valueToAdd := lua.CheckMap(l, 2)
```

Comment:
Similar to CheckNumber there exists CheckTable(), however that only checks the type. Can we retrieve the value using next() or any other similar function?